### DISCUSSION TITLE
Update-DbaInstance  - skip required reboot

### DISCUSSION THREAD

**Discussion by chikenborgir** [dataplat/dbatools#9059] [Q&A] [answered]
Hi there,

since SQL Server 2019 CU21 I'm having a issue with the [Update-DbaInstance](https://docs.dbatools.io/Update-DbaInstance.html) command using `ArgumentList = "/SkipRules=RebootRequiredCheck"`

<br>

Error message: `WARNING: [00:00:00] hostname.domain is pending a reboot. Reboot the computer before proceeding.`

<br>

Tested dbatools version: `1.1.146 / 2.0.4`

<br>

Anyone aware what causes this? It worked before SQL Server 2019 CU21 without any issues. Thanks in advance.

<br>

PowerShell script used for updating the instance directly after the installation:
```powershell
[CmdLetBinding()]Param(
    [Parameter()][string]$UpdatePath = "C:\SQLSERVER\UPDATES",
    [Parameter()][string]$ExtractPath = "C:\SQLSERVER\UPDATES\TEMP",
    [Parameter(Mandatory)][int32[]]$KB
)

$ErrorActionPreference = "Stop"
$WarningPreference = "Stop" 

Try {
    Import-Module dbatools

    $SplatParameters = @{
        ArgumentList = "/SkipRules=RebootRequiredCheck"
        ComputerName = $env:COMPUTERNAME
        KB = $KB
        Path = $UpdatePath
        ExtractPath = $ExtractPath
        Confirm = $false
    }

    Update-DbaInstance @SplatParameters
} Catch {
    Write-Output $_
    exit 1
}
```

**Comment by corythosaurus** [comment]
I'm not sure if this is your issue, but in my case I found that there were files pending rename. If you go in and rename that registry key it returns false. There is "[-NoPendingRenameCheck]" but it doesnt seem to actually do anything.

```
Install-Module -Name PendingReboot
Test-PendingReboot -Detailed
```

**Comment by chikenborgir** [comment]
I resolved this issue by using Start-Process of the KB directly instead of using dbatools.

**Comment by chikenborgir** [comment]
I reopened the discussion since I'm still having the issue and got some time to test a manual patch. Also, I want to understand it incase I'm doing anything wrong since we prefer dbatools.

<br>

I tested the following 
```powershell
# not working (dbatools)
$splatParameter = @{
    NoPendingRenameCheck = $true
    ComputerName = $env:COMPUTERNAME
    Path = "C:\SqlServer\Update\"
    ExtractPath = "C:\SqlServer\Update\Temp"
    KB = 5031778
    Confirm = $false
}
Update-DbaInstance @splatParameter

# not working - trying to use same ArgumentList as Start-Process (dbatools)
$ArgumentList = @("/SkipRules=RebootRequiredCheck", "/IACCEPTSQLSERVERLICENSETERMS", "/SUPPRESSPAIDEDITIONNOTICE", "/SUPPRESSPRIVACYSTATEMENTNOTICE", "/ACTION=PATCH", "/INSTANCENAME=MSSQLSERVER", "/QUIET")

$splatParameter = @{
    NoPendingRenameCheck = $true
    ComputerName = $env:COMPUTERNAME
    Path = "C:\SqlServer\Update\"
    ExtractPath = "C:\SqlServer\Update\Temp"
    ArgumentList = $ArgumentList
    KB = 5031778
    Confirm = $false
}
Update-DbaInstance @splatParameter


# working (SQLServer2022-KB5031778-x64.exe)
$splatParameters = @{
    FilePath = "C:\SqlServer\Update\SQLServer2022-KB5031778-x64.exe"
    ArgumentList = "/SkipRules=RebootRequiredCheck /IACCEPTSQLSERVERLICENSETERMS /SUPPRESSPAIDEDITIONNOTICE /SUPPRESSPRIVACYSTATEMENTNOTICE /ACTION=PATCH /INSTANCENAME=MSSQLSERVER /QUIET"
    PassThru = $true
    Wait = $true
}

Start-Process @splatParameters
```
<br>
<br>
Output:
<br>

With dbatools I can't get the patch working. I always receive "Reboot the computer before proceeding"

![dbatools](https://github.com/dataplat/dbatools/assets/20847376/47e44114-9fc4-4cb9-ac2f-5675bc970b0e)

<br>
Without dbatools it's patching without any issues

![patch-executeable](https://github.com/dataplat/dbatools/assets/20847376/0ad94487-491b-4745-8c6f-272ac940488c)

**Comment by wsmelton** [comment]
You are seeing that message because the command itself checks for reboot status, it does not acknowledge the argument you are passing as that is strictly for the patch executable flags.

https://github.com/dataplat/dbatools/blob/b36ddcf68783c30de8c1f32306ca16928e22b32c/public/Update-DbaInstance.ps1#L437-L445

**Reply by wsmelton** [reply]
Which we don't seem to have touched that file in a long time so the codes we are checking may be outdated or even changed in Windows 😆 
https://github.com/dataplat/dbatools/blob/development/private/functions/Test-PendingReboot.ps1

**Comment by chikenborgir** [comment]
Thanks for your fast response. So is there no way to skip the required reboot check using Update-DbaInstance? As mentioned in my first post we started facing this issue since SQL Server 2019 CU21 but the -NoPendingRenameCheck parameter seems not to help here and patching with the executable directly (Start-Process) works without any issues (at least without rebooting 😅)

<br>

We have customers who insist on hardware servers that perform a memory test for about 5-10 minutes when rebooting, so an extra reboot for these servers would greatly increase the downtime.

**Comment by chikenborgir** [comment, marked as answer]
Since no solution was provided and to close the discussion I'm patching the instances now by executing the KB directly instead of using dbatools command Update-DbaInstance.

It works fine, better and without issues.

**Reply by wsmelton** [reply]
There is an active bug report for the parameter `NoPendingRenameCheck` not functioning correct.
Ref #9219
